### ISSUE TITLE
Add a check for bare [[ ]] and (( ))  in bats tests (with pre 4.1 bash)

### ISSUE THREAD

**Issue by martin-schulze-vireso** [koalaman/shellcheck#2360]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bats

@test test {
        [[ $BASH_VERSION == no-match ]] # this should fail the test
        true # this "shadows" the error return value
}
```

Run with bash 4.1:
```
docker run --rm -v $PWD:/bats-core bash:4.1 /bats-core/bin/bats /bats-core/test.bats
1..1
not ok 1 test
# (in test file /bats-core/test.bats, line 4)
#   `[[ $BASH_VERSION == no-match ]] # this should fail the test' failed
```

Run with bash 4.0:
```
$ docker run --rm -v $PWD:/bats-core bash:4.0 /bats-core/bin/bats /bats-core/test.bats
1..1
ok 1 test
```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

```
        [[ $BASH_VERSION == no-match ]] 
        ^-- SC20xx: This expression won't fail the test on Bash 4.0 or older
```

A possible fix is appending `|| false` to the line.
The most likely victim of this problem would be MacOS users who still run Bash 3.
The Bash version dependency makes this a bit harder. I am still toying with the idea of mandating a higher minimum version (which would exclude MacOS users) or adding a directive that allows users to restrict the bash version themselves, thus allowing to only give this message when there is no minimum version or the version is below 4.1. This limit would have to be incorporated into this check.

See https://github.com/sstephenson/bats/issues/49 for related discussion.

**Comment by giner** [comment]
Here is more context and details.

Bash was not applying `set -e` for `[[ ... ]]` and `(( ... ))` before version 4.1, see https://github.com/bminor/bash/blob/637f5c8696a6adc9b4519f1cd74aa78492266b7f/CHANGES#L5496.

```bash
for bash_version in 3.2 4.0 4.1; do
  docker run --rm -it "bash:$bash_version" -c '
    set -e
    echo "BASH_VERSION: $BASH_VERSION"
    echo "Before f()"
    f() { [[ 1 -eq 2 ]]; true; }; echo "$(declare -f f | tr -d "\n")"; f
    echo "After f()"
    echo
  '
done
```
```bash
BASH_VERSION: 3.2.57(1)-release
Before f()
f () {     [[ 1 -eq 2 ]];    true}
After f()

BASH_VERSION: 4.0.44(1)-release
Before f()
f () {     [[ 1 -eq 2 ]];    true}
After f()

BASH_VERSION: 4.1.17(2)-release
Before f()
f () {     [[ 1 -eq 2 ]];    true}
```

```bash
for bash_version in 3.2 4.0 4.1; do
  docker run --rm -it "bash:$bash_version" -c '
    set -e
    echo "BASH_VERSION: $BASH_VERSION"
    echo "Before f()"
    f() { (( 1 == 2 )); true; }; echo "$(declare -f f | tr -d "\n")"; f
    echo "After f()"
    echo
  '
done
```
```bash
BASH_VERSION: 3.2.57(1)-release
Before f()
f () {     (( 1 == 2 ));    true}
After f()

BASH_VERSION: 4.0.44(1)-release
Before f()
f () {     (( 1 == 2 ));    true}
After f()

BASH_VERSION: 4.1.17(2)-release
Before f()
f () {     (( 1 == 2 ));    true}
```

macOS is using Bash 3.2 by default which makes many want to support this version of Bash, which also does [bats](https://github.com/bats-core/bats-core) project. This issue often gets unnoticed and in case of bats it [results](https://github.com/bats-core/bats-core/issues/1178) in false-positive test results. There seem to be no easy solution on the bats side and it covering it by ShellCheck would help a lot.
